src/renderer/hooks/useAnimationEnd.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+/**
+ * A custom hook to implement an animationend listener on the provided ref
+ * @param {object} ref - A reference to a DOM element to add the listener to
+ * @param {function} callback - A callback function that will be run for matching animation events
+ * @param {string} animationName - The name of the animation to listen for
+ * @param {boolean} listenForEventBubbling - A parameter that when true, listens for events on the target element and
+ *   bubbled from all descendent elements but when false, only listens for events coming from the target element and
+ *   ignores events bubbling up from descendent elements
+ */
+function useAnimationEnd(
+    ref,
+    callback,
+    animationName,
+    listenForEventBubbling = true,
+) {
+    React.useEffect(() => {
+        if (!ref.current) {
+            return undefined;
```

Comment:
I think `undefined` can't be reasigned anymore, but just in case
```suggestion
            return;
```

Reply:
That breaks the linter, since we're returning either `void` or a function. Returning `undefined` is a different type and denotes that the function always returns something (in this case, nothing)